frontend/src/views/Game.tsx

```diff
@@ -152,76 +75,28 @@ function GamePage() {
     setAllSolved(newGame.allSolved);
     setTimeUp(newGame.gameTimer.timeUp);
     setGameEnded(newGame.gameEnded);
-  };
-
-  const dispatch = useAppDispatch();
-  const { currentUser, game } = useAppSelector((state) => state);
+    setStartTime(newGame.gameTimer.startTime);
 
-  const setDefaultCodeFromProblems = useCallback((problemsParam: Problem[],
-    code: string, language: Language) => {
-    const promises: Promise<DefaultCodeType>[] = [];
-    problemsParam.forEach((problem) => {
-      if (problem && problem.problemId) {
-        promises.push(getDefaultCodeMap(problem.problemId));
+    // Set the spectator list in the game.
+    const spectatorList: User[] = [];
+    newGame.room.users.forEach((user) => {
+      if (user.spectator) {
+        spectatorList.push(user);
       }
     });
+    setSpectators(spectatorList);
```

Comment:
I considered this - initially I didn't do that on this PR mainly just to avoid adding further complexity. However, since we have an active and inactive users list, I think it makes to add spectators as well, so I'll do that next.

Reply:
Done! ✅